source/beamline/CLS/CLSAMDSScaler.h

```diff
@@ -0,0 +1,318 @@
+	AMDetectorDefinitions::ReadMode readModeForTriggerSource_;
+
+	/// The common dwell time source for this system. Detector implementations can return this as a common means for triggering and comparing shared triggers.
+	AMDetectorDwellTimeSource *dwellTimeSource_;
+	bool dwellTimeSourceRequested_;
+
+
+	/// the AMDS Amptek Server identifier
+	QString amdsServerIdentifier_;
+	/// The AMDS buffer name for this instance
+	QString amdsBufferName_;
+
+	/// The data returned from the last acquire(AMDetectorDefinitions::ContinuousMode) call
+	AMDSClientRelativeCountPlusCountDataRequest *lastContinuousDataRequest_;
+	/// The window for the continuous data request in seconds
+	double continuousDataWindowSeconds_;
+	/// The underlying polling rate of the AMDS implementation, usually 1ms
+	int pollingRateMilliSeconds_;
+};
+
+/// This class is an abstraction of an individual channel for the scaler class.
+class CLSAMDSScalerChannel : public QObject
+{
+Q_OBJECT
+
+public:
+	/// Constructor.  Takes in a base name and the index of the channel and builds all the necessary PV's.
+	CLSAMDSScalerChannel(const QString &baseName, int index, QObject *parent = 0);
+	virtual ~CLSAMDSScalerChannel();
+
+	/// Returns whether the channel is connected.
+	bool isConnected() const;
+	/// Returns whether the channel is enabled.
+	bool isEnabled() const;
+	/// Returns the current value of the channel.
+	int reading() const;
+	/// Returns the current voltage.  \note Not high voltage.
+	double voltage() const;
+	/// Returns the index of this channel.
+	int index() const { return index_; }
+
+	/// Return 0 right now, we haven't done the enable/disable portion yet
+	/// Creates an action that will enable/disable the channel based on \param setEnabled.
+	AMAction3* createEnableAction3(bool setEnabled);
+
+	/// Returns the custom settable name
+	QString customChannelName() const { return customChannelName_; }
+
+	/// Sets a SR570 to this particular channel.  It is assumed that the SR570 ACTUALLY corresponds to the scalar channel, otherwise, this is meaningless.
+	void setCurrentAmplifier(AMCurrentAmplifier *amplifier);
+	/// Returns the SR570 that this scaler channel uses for sensitivity changes.  Returns 0 if not set.
+	AMCurrentAmplifier *currentAmplifier() const { return currentAmplifier_; }
+
+	/// Returns the AMDetector that this scaler channel uses.
+	AMDetector* detector() const { return detector_; }
+
+	/// Returns the minimum voltage range for the ion chamber.
+	double minimumVoltage() const { return voltageRange_.minimum(); }
+	/// Returns the maximum voltage range for the ion chamber.
+	double maximumVoltage() const { return voltageRange_.maximum(); }
+	/// Returns the voltage range for the ion chamber.
+	const AMRange &voltageRange() const { return voltageRange_; }
+
+	/// Returns whether the voltage is within the linear range of the ion chamber.
+	virtual bool withinLinearRange() const { return voltage() <= maximumVoltage() && voltage() >= minimumVoltage(); }
```

Comment:
You could have used the AMRange API for this: `return voltageRange_.withinRange(voltage());`